types/src/storage.rs

```diff
@@ -459,6 +461,29 @@ impl<T: RocksDB> KVStorage for T {
         }
     }
 
+    fn count(&self, from_key: &str, to_key: &str) -> Result<usize> {
+        let direction = if from_key <= to_key {
+            rocksdb::Direction::Forward
+        } else {
+            rocksdb::Direction::Reverse
+        };
+        let iter = self.iterator(IteratorMode::From(from_key.as_bytes(), direction));
+
+        let mut count = 0;
+
+        for item in iter {
+            let (k, _value) = item?;
+            if matches!(direction, rocksdb::Direction::Forward) && k.as_ref() > to_key.as_bytes()
+                || matches!(direction, rocksdb::Direction::Reverse)
+                    && k.as_ref() < to_key.as_bytes()
+            {
+                break;
+            }
+            count += 1;
+        }
```

Comment:
You could use the `iterate_raw` here:
```rust
    fn count(&self, from_key: &str, to_key: &str) -> Result<usize> {
        let mut count = 0;

        for res in self.iterate_raw(from_key, to_key) {
            let _ = res?;
            count += 1;
        }
        Ok(count)
    }
```

and implement it on the `KVStorage` trait directly (as an already provided method).